packages/editor/src/store/selectors.js

```diff
@@ -802,6 +802,29 @@ export const isSavingPost = createRegistrySelector( ( select ) => ( state ) => {
 	);
 } );
 
+/**
+ * Returns true if non-post entities are currently being saved, or false otherwise.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} Whether non-post entities are being saved.
+ */
+export const isSavingNonPostEntityChanges = createRegistrySelector(
```

Comment:
I think that selector name is a bit confusing, it doesn't match what it's actually doing. Because in theory, you could be saving other posts (think query block) that are different from the main one being edited in the page.

So it's more like

`isSavingExtraEntities` or `isSavingMultipleEntities` or something like that

Reply:
Makes me wonder though whether we should just check that we're saving any entity including the "post" one. I mean do we need the distinction in the component where we call the selector? Maybe the selector could be simpler if we just check whether there's an entity save in progress (regardless of which entity it is)

Reply:
> I think that selector name is a bit confusing, it doesn't match what it's actually doing. Because in theory, you could be saving other posts (think query block) that are different from the main one being edited in the page.
>
> So it's more like
>
> `isSavingExtraEntities` or `isSavingMultipleEntities` or something like that

That's true, but I modeled `isSavingNonPostEntityChanges` pretty much after `hasNonPostEntityChanges`, and the same points could be made about that selector, no? So I thought it'd make sense to name the "is-saving" counterpart to the "has...changes" selector similarly...

https://github.com/WordPress/gutenberg/blob/0216850487587c6635397e54f0bcd6e297103208/packages/editor/src/store/selectors.js#L150-L172

Reply:
Any thoughts about my second comment here, it makes the first one irrelevant because maybe the selector is not needed at all

Reply:
> Makes me wonder though whether we should just check that we're saving any entity including the "post" one. I mean do we need the distinction in the component where we call the selector?

It seems like in the context of the `PostPublishButton`, that would be fine (since it's all going into `isButtonDisabled` and `isToggleDisabled` anyway). About other contexts, I'm less sure. Even within the `PostPublishPanel`, we have such props as `isSaving` and `isSaveable` (connected to the `isSavingPost()` and `isEditedPostSaveable()` selectors, respectively), but also `forceIsSaving` 😖  So that's not something I'm eager to touch 😬 

Now if we're going to keep `isSavingPost()` in a number of other contexts (at least for now), it does seem to make sense to me to have a selector that's complementary to it  -- i.e. checks if we're saving changes to anything _but_ the post that's currently being edited.

In the long run, it might make sense to replace `isSavingPost()` by a wholesale `isSavingEntities()` selector, but that'll require an audit of all callsites. For the time being, I think it would add to the confusion to use such a wholesale selector for "is-saving" detection, when OTOH we have a dedicated `hasNonPostEntityChanges` for "is-dirty" detection.

> Maybe the selector could be simpler if we just check whether there's an entity save in progress (regardless of which entity it is)

FWIW, that would basically just be [`__experimentalGetEntitiesBeingSaved()`](https://github.com/WordPress/gutenberg/pull/33140/files#diff-b92e722dcdf42bb6b4bece6f3d64e34fa386361c5606f1c6a47b3dd7ac03bee1R326), right? Which is the counterpart to `__experimentalGetDirtyEntityRecords()`.

Reply:
That makes sense to me yeah. I have a feeling that our saving flow UI bits are way too complex and very hard to maintain for what they really are meant to do. they are also very old components that didn't see a lot of improvements over time aside the addition of adhoc multi-entities saving flow related changes but without rethinking the components themselves . I do think we need to check there for simplification/refactoring opportunities soon.

Reply:
Indeed! Coincidentally, here's a related UX issue: https://github.com/WordPress/gutenberg/issues/33223